module/icp/io/sha2_mod.c

```diff
@@ -623,6 +623,20 @@ sha2_digest_atomic(crypto_provider_handle_t provider,
 
 	if (mechanism->cm_type <= SHA256_HMAC_GEN_MECH_INFO_TYPE)
 		sha_digest_len = SHA256_DIGEST_LENGTH;
+	else if (mechanism->cm_type <= SHA384_HMAC_GEN_MECH_INFO_TYPE)
+		sha_digest_len = SHA384_DIGEST_LENGTH;
+	else if (mechanism->cm_type <= SHA512_HMAC_GEN_MECH_INFO_TYPE)
+		sha_digest_len = SHA512_DIGEST_LENGTH;
+	else if (mechanism->cm_type == SHA512_224_MECH_INFO_TYPE)
+		sha_digest_len = SHA512_224_DIGEST_LENGTH;
+	else if (mechanism->cm_type == SHA512_256_MECH_INFO_TYPE)
+		sha_digest_len = SHA512_256_DIGEST_LENGTH;
+	else {
+		/* Bad crypto mechanism */
+		ret = CRYPTO_MECHANISM_INVALID;
+		digest->cd_length = 0;
+		return (ret);
+	}
```

Comment:
The correct fix for this is in #5251 along with some other fixes for sha2_mod.c. Please remove it from here so we can deal with it there. The rest LGTM.